### PR TITLE
docs: cloud infrastructure

### LINKED ISSUE #30: Add Cloud Infra Docs

### PR DESCRIPTION
**Pull request by NguyenPatrick**
Closes #30

### CODE REVIEW

**Review by AnthonyLaiRH — APPROVED** [review]
lgtm

File: docs/design/cloud_infra.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- It would very beneficial to be able have the terraform commands be automated so one doesn't have to manually run the commands locally
+
+### Tasks:
+
+[ ] Store the terraform state in [Google Cloud Storage](https://cloud.google.com/storage)
+[ ] Define terraform repository strucuture
+[ ] Determine the terraform modules to use
+[ ] (Potential) Setup [Atlantis](https://www.runatlantis.io/) to run Terraform commands from Github and keep a history of the output logs
+
+## The K8s Route
```
Comment (callmekungfu) [inline docs/design/cloud_infra.md:41]: I think for the first iteration we can deploy the service on a VM, or Google App Engine. We can consider k8s if we need to orchestrate a lot of containers.

File: docs/design/cloud_infra.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Cloud Infrastructure
+
+The entire infrastructure of the project will be hosted on GCP
+
+- Secrets: GCP Secrets Manager
+  - Manually store sensitive information
+- Kubernetes Cluster: GKE (Google Kubernetes Engine)
+- Images: GCR (Google Container Registry)
+
```
Comment (callmekungfu) [inline docs/design/cloud_infra.md:9]: Can you recommend a plan for hosting our database (MongoDB)? Do you think we should host our own instance on a VM or should we use Atlas?

**Review by callmekungfu — COMMENTED** [review]
Added some thoughts on business layer deployment for the first layer and question on db hosting

**Pull request merged**